rfcs/rdm-0079-collections.md

```diff
@@ -0,0 +1,337 @@
+- Start Date: 2024-09-03
+- RFC PR: [#87](https://github.com/inveniosoftware/rfcs/pull/87)
+- Authors: Alex, Carlin, Fatimah, Manuel
+
+# Community collections
+
+## Summary
+
+To describe collections, lets use the following example records:
+    
+```yaml
+Policy Brief for iMagine        <- Record
+    Subjects:
+        Ecosystems
+    Funding: 
+        Award:
+            Title: iMagine
+            Program: Horizon Europe
+
+Super-Resolved SAR Data         <- Record
+    Subjects
+        Artificial Intelligence
+    Funding: 
+        Award:
+            Title: S34I
+            Program: Horizon Europe
+```
+
+In *collections*, these records would be grouped (or filtered) as follows
+
+```text
+Subjects                          <- CollectionTree
+    Artificial Intelligence       <- Collection
+        Super-Resolved SAR Data   <- Record
+    Ecosystems                    <- Collection
+        Policy Brief for iMagine  <- Record
+
+Funding Programmes                <- CollectionTree
+    Horizon Europe                <- Collection
+        Policy Brief for iMagine  <- Record
+        Super-Resolved SAR Data   <- Record
+```
+                                     
+For instance, collections can provide valuable insights of the contents of a community based on its records metadata, e.g. the funding metadata. This provides a more unique way of navigate through communities and analyse its records.
+
+## Motivation
+
+- As a community manager, I want to know how many records of my community are funded by the same grant
+- As a community manager, I want a high-level view that provides insights on the community records
+- As a researcher, I want to see all the records which are in a subject
+
+## Detailed design
+                                     
+### Key points 
+                                     
+In summary, below are the key points of this design along with the locations of the relevant discussions:
+
+- [🔑 Database considerations](#ctrees)
+- [🔑 Indices](#indices)
+- [🔑 Queries composition](#queries)
+                                     
+### Design
+    
+At a glance, a collection can be thought of as a set of records. To do so, we further expand the definition into:
```

Comment:
A key point is that collections are **queries** over all records. A "set of records" I understand as an explicit assignment of a record to a collection which is what communities is.

Reply:
maybe adding "transversal queries" would help?